File: backend/app/schemas/article.py
"""文章相关 Pydantic 模型。"""
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field

from app.schemas.tag import TagOut


class ArticleCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    summary: str = Field("", max_length=500)
    content: str
    cover_image: str = ""
    category_id: Optional[int] = None
    tags: List[str] = Field(default_factory=list)


class ArticleUpdate(BaseModel):
    title: Optional[str] = Field(None, max_length=200)
    summary: Optional[str] = Field(None, max_length=500)
    content: Optional[str] = None
    cover_image: Optional[str] = None
    category_id: Optional[int] = None
    tags: Optional[List[str]] = None


class ArticleDetail(BaseModel):
    id: int
    title: str
    summary: str
    content: str
    content_html: str
    cover_image: str
    category: Optional[dict] = None
    tags: List[TagOut] = Field(default_factory=list)
    view_count: int
    like_count: int
    status: int
    published_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime


class ArticleListItem(BaseModel):
    id: int
    title: str
    summary: str
    cover_image: str
    category_id: Optional[int]
    view_count: int
    like_count: int
    status: int
    published_at: Optional[datetime]
    created_at: datetime

File: frontend/src/types/article.ts
import type { Category, Tag } from './api'

export interface ArticleAuthor {
  id: number
  nickname: string
  avatar: string
}

export interface Article {
  id: number
  title: string
  summary: string
  content: string
  content_html?: string
  cover_image?: string
  category?: Category | null
  tags?: Tag[]
  view_count: number
  like_count: number
  status: number
  published_at?: string | null
  created_at: string
  updated_at: string
  // 博客系统新增互动字段（详情接口返回）
  author?: ArticleAuthor | null
  comment_count?: number
  share_count?: number
  is_liked?: boolean
  is_favorited?: boolean
}

export interface ArticleCreatePayload {
  title: string
  summary?: string
  content: string
  cover_image?: string
  category_id?: number | null
  tags?: string[]
}

export interface ArticleListParams {
  page?: number
  page_size?: number
  status?: number
  keyword?: string
}

export interface LikeResult {
  liked: boolean
  like_count: number
}
